File: noaa-requests/matlab/NOAA_Water_Level_Downloader/WL_Downloader_Functions/download_segmentation.m
function [stDates,endDates,stDatesp,endDatesp] = download_segmentation(dStruct,flag1,indx)    
%% DIVIDE DATE RANGE IN ALLOWED SEGMENTS
    if strcmp(flag1,'m')==0
        % Water Level measurements segmentation
        [stDates,endDates] = date_range_segmentation(dStruct,flag1,indx);
    else
        % Monthly data does not have a data download limit hence date segmentation
        % is not necessary
        % Get date ranges
        stDates = dStruct.startDate(indx);
        endDates = dStruct.endDate(indx);
        % Cut date strings
        stDates = cellfun(@(x) x(1:19), stDates, 'un', 0);
        endDates = cellfun(@(x) x(1:19), endDates, 'un', 0);
        % Reformat date ranges for API
        stDates = datestr(datenum(stDates,'yyyy-mm-dd HH:MM:SS'),'yyyymmdd HH:MM');
        endDates = datestr(datenum(endDates,'yyyy-mm-dd HH:MM:SS'),'yyyymmdd HH:MM');
    end
    
    % Tidal predictions segmentation
    [stDatesp,endDatesp] = date_range_segmentation_predictions_V2(dStruct,indx);
    %         [stDatesp,endDatesp] = date_range_segmentation_predictions(dStruct,indx);
end

File: noaa-requests/noaa-py/noaapy/download_segmentation.py
# noaa-requests/noaa-py/noaapy/download_segmentation.py
import datetime

import noaapy


def download_segmentation(d_struct, flag1, indx):
    if flag1 != "m":
        st_dates, end_dates = noaapy.date.date_range_segmentation(d_struct, flag1, indx)
    else:
        st_dates = d_struct["start_date"][indx]
        end_dates = d_struct["end_date"][indx]
        st_dates = [date[:19] for date in st_dates]
        end_dates = [date[:19] for date in end_dates]
        st_dates = [
            datetime.datetime.strptime(date, "%Y-%m-%d %H:%M:%S").strftime(
                "%Y%m%d %H:%M"
            )
            for date in st_dates
        ]
        end_dates = [
            datetime.datetime.strptime(date, "%Y-%m-%d %H:%M:%S").strftime(
                "%Y%m%d %H:%M"
            )
            for date in end_dates
        ]

    st_dates_p, end_dates_p = noaapy.date.date_range_segmentation_predictions_v2(
        d_struct, indx
    )

    return st_dates, end_dates, st_dates_p, end_dates_p
